### ISSUE TITLE
Separate resources section for leaders and followers for redis-cluster chart

### ISSUE THREAD

**Issue by AharonIsraelCohen** [OT-CONTAINER-KIT/redis-operator#1186]
**Is your feature request related to a problem? Please describe.**
I tried to update my cluster resources, I am using operator and cluster latests charts, and I noticed that there is a Rolling update 
action for the leaders and right after it finished the followers starts to update as well, as a result we lost some data because 
the leaders didn't managed to catch up with the followers and then the followers started to restart as well/

**Describe the solution you'd like**
I would like to have separate "resource" section for leaders and followers, so we could update them seperatly.

**What version of redis-operator are you using?**
chart version: 0.18.5
image:

**Comment by husnialhamdani** [comment]
Hi @AharonIsraelCohen, 
You mean, ability to have separate "resource" (CPU/Memory requests/limits)?

@drivebyer you can assign me

**Comment by AharonIsraelCohen** [comment]
> Hi @AharonIsraelCohen, You mean, ability to have separate "resource" (CPU/Memory requests/limits)?
> 
> @drivebyer you can assign me

Exactly, so on cases like increasing resources we could increase follower resources first and after the cluster is balanced, to increase leaders resources as well.

**Comment by drivebyer** [comment]
After thinking of this, I think we didn't need an additional field to separate resources for leader and follower. Because the actual role of Redis will change after cluster failover. That is to say, the pod under the leader statefulset may become a follower. So the additional field will be set to the wrong pod after cluster failover.

In your case, I think we need another way to enhance the data security. @AharonIsraelCohen 

cc @husnialhamdani

**Comment by husnialhamdani** [comment]
> After thinking of this, I think we didn't need an additional field to separate resources for leader and follower. Because the actual role of Redis will change after cluster failover. That is to say, the pod under the leader statefulset may become a follower. So the additional field will be set to the wrong pod after cluster failover.
> 
> In your case, I think we need another way to enhance the data security. @AharonIsraelCohen
> 
> cc @husnialhamdani

In my opinion it's still makes sense, to support resiliency/availability to the cluster, like we can make resources adjusment on the fly, by updating the follower first and then the leader (vice versa), follower can be temporarily become leader and then leader can takeover back after we patched the resources.
